src/wallet/rpcwallet.cpp

```diff
@@ -2737,7 +2951,21 @@ UniValue getwalletinfo(const UniValue& params, bool fHelp)
     obj.push_back(Pair("immature_cold_staking_balance",    ValueFromAmount(pwalletMain->GetImmatureColdStakingBalance())));
     obj.push_back(Pair("txcount", (int)pwalletMain->mapWallet.size()));
     obj.push_back(Pair("keypoololdest", pwalletMain->GetOldestKeyPoolTime()));
-    obj.push_back(Pair("keypoolsize", (int)pwalletMain->GetKeyPoolSize()));
+
+    size_t kpExternalSize = pwalletMain->KeypoolCountExternalKeys();
+    obj.pushKV("keypoolsize", (int64_t)kpExternalSize);
+
+    ScriptPubKeyMan* spk_man = pwalletMain->GetScriptPubKeyMan();
+    if (spk_man) {
+        const CKeyID& seed_id = spk_man->GetHDChain().GetID();
+        if (!seed_id.IsNull()) {
+            obj.pushKV("hdseedid", seed_id.GetHex());
+        }
+    }
+    if (pwalletMain->CanSupportFeature(FEATURE_HD_SPLIT)) {
+        obj.pushKV("keypoolsize_hd_internal",   (int64_t)(pwalletMain->GetKeyPoolSize() - kpExternalSize));
+    }
```

Comment:
Shouldn't we check that `IsHDEnabled()` rather than `CanSupportFeature(FEATURE_HD_SPLIT)`?
Also we should display `setStakingKeyPool` size too somewhere.

Reply:
done